keras/utils/generic_utils.py

```diff
@@ -328,7 +328,18 @@ def update(self, current, values=None, force=False):
                 time_per_unit = 0
             if self.target is not None and current <= self.target:
                 eta = time_per_unit * (self.target - current)
-                info = ' - ETA: %0.fs' % eta
+                hours = eta // (60 * 60)
+                minutes = (eta % (60 * 60)) // 60
```

Comment:
Replacing `60 * 60` with `3600` here and above would save quite a bit of computation, since this method is called a lot.

Reply:
Actually, maybe it would be higher performance to have a switch:

`if eta > 3600` / `if eta > 60`

So that we only compute what we need.

Reply:
Ah right, used to c++ optimizations I guess. Changed as per request.